src/main/java/com/minecolonies/coremod/colony/ColonyManager.java

```diff
@@ -992,49 +1008,128 @@ public static void onWorldLoad(@NotNull final World world)
+            Log.getLogger().info(String.format("Initiating scheduled backup"));
+            backupColonyData();
+        }
+    }
 
-            for (int i = 1; i < colonies.getTopID() + 1; i++)
+    /**
+     * Thread safe method to lock backups so only one backup may run at a time
+     *
+     * @return boolean indicating whether safe to initiate backup
+     */
+    synchronized private static boolean isSafeToBackup()
+    {
+        if (backupInProgress)
+        {
+            Log.getLogger().warn("Cannot initiate backup - already in progress");
+            return false;
+        }
+        else
+        {
+            backupInProgress = true;
+            return true;
+        }
+    }
+
+    public static boolean backupColonyData()
+    {
+        // Possibility of the timer kicking in while the command backup is being run, or vice versa
+        // Prevent by requiring a threadsafe token, reject any other backup attempts while the token
+        // is checked out - why run two backups back to back?
+        if (isSafeToBackup())
+        {
+            // Cancel any pending scheduled backup - we'll reset the timer after we're done
+            if (backupTimerTask != null)
             {
-                @NotNull final File file = new File(saveDir, String.format(FILENAME_COLONY, i));
-                if (file.exists())
+                backupTimerTask.cancel();
+            }
+
+            if (numWorldsLoaded > 0 && saveNeeded)
+            {
+                saveColonies(false);
+            }
+
+            final File backupSaveLocation = getBackupSaveLocation(new Date()); // todo Stop taking a gamble that the filename doesn't already exist, it's possible
+            // Just in case, try to create the parent backup directory if it doesn't exist
+            backupSaveLocation.getParentFile().mkdir();
+
+            try (FileOutputStream fos = new FileOutputStream(backupSaveLocation))
+            {
+                @NotNull final File saveDir =
+                  new File(DimensionManager.getWorld(0).getSaveHandler().getWorldDirectory(), FILENAME_MINECOLONIES_PATH); //todo remove hardcoded world 0
+                final ZipOutputStream zos = new ZipOutputStream(fos);
+
+                for (final Colony backupTargetColony : colonies)
                 {
-                    addToZipFile(String.format(FILENAME_COLONY, i), zos, saveDir);
+                    @NotNull final File file = new File(saveDir, String.format(FILENAME_COLONY, backupTargetColony.getID()));
+                    if (file.exists())
+                    {
+                        addToZipFile(file, zos);
+                    }
                 }
-            }
-            addToZipFile(getSaveLocation().getName(), zos, saveDir);
+                addToZipFile(getSaveLocation(), zos);
 
-            zos.close();
-            fos.close();
+                zos.close();
+                fos.close();
+            }
+            catch (final Exception e)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)